### DISCUSSION TITLE
March England contour update less detailed?

### DISCUSSION THREAD

**Discussion by thompehw** [osmandapp/OsmAnd#11130] [General]
The March 2021 update to England's contour lines has recently been released. Here is how it shows a place I know:
![Screenshot_20210313-031846_OsmAnd+](https://user-images.githubusercontent.com/39751382/111017818-788cb400-83ad-11eb-867e-f926044ce7f4.jpg)

The area without trees around the viewpoint  (50.8886693, -0.0107583 ) is flat, cut into the hillside by an old quarry. The March update for England's contours isn't accurate enough to show this, and that was also how it was in OsmAnd long ago.

But the contours file (October 2020?) that is replaced was better:
![Screenshot_20210313-031637_OsmAnd+](https://user-images.githubusercontent.com/39751382/111017990-665f4580-83ae-11eb-9fdd-ae20011fd8eb.jpg)

I could give other examples for places nearby.

I would like to understand. My guess would be that in 2020 there was an experiment to use England Environment Agency LIDAR data to augment SRTM, and now that has been abandoned?

Thanks for any tips!

**Comment by xmd5a2** [comment]
This is an error. I've created issue for this https://github.com/osmandapp/OsmAnd/issues/11152

**Reply by thompehw** [reply]
Thank you! I see there is a new contour file available today which is already better.

**Reply by thompehw** [reply]
And double thank-you, @xmd5a2, for taking the time to give such an excellent explanation of smoothing in #11152. The sand dunes example convincingly shows  how essential it is.

I was very interested to read that your source is [Digital LIDAR-Terrain Models of Europe](https://data.opendataportal.at/dataset/dtm-europe). I had been puzzled how England contours were now so good if you were only using SRTM.

It was extraordinarily generous to offer to generate custom contours for me! But I could do it myself. When OsmAnd was just using SRTM I used QGIS to generate some OsmAnd contours combining our patchy LIDAR with SRTM as your source now does. In my location I didn't have a problem with the visual aspects of smoothing, but file size was an issue (I was being ridiculously precise, taking the very best LIDAR data for each spot - it varies here, because the data is gathered for flood modelling).

If any other fans of England LIDAR are reading this, check out the thread surrounding [Talk-GB Environment agency LIDAR tiles](https://lists.openstreetmap.org/pipermail/talk-gb/2021-March/026354.html). There are tile servers you can use in Vespucci or JOSM for OSM mapping (the OGL license only requires attribution).

**Comment by tommycrock** [comment]
All very interesting. I was wondering if the handling of the smoothing mentioned in #11152 could be improved to reduce the information loss in hilly/rocky areas.

For example, 
![1m LIDAR-ringed](https://user-images.githubusercontent.com/5749884/111898718-a4e4a800-8a1f-11eb-9309-e279de23915a.PNG)
The Environment Agency 1m DTM tiles show a clear quarry and cutting that I've ringed here. In the DTM-Europe dataset (1", .hgt) these show up, albeit lower resolution (I applied cubic resampling in QGIS and then massively altered the contrast in GIMP to show):
![e-DTM-ringed](https://user-images.githubusercontent.com/5749884/111899163-b11e3480-8a22-11eb-80a5-843fa1a14562.PNG)
but, in osmand contours, the quarry has disappeared and the cutting only shows one side:
![Screenshot_OsmAnd+ringed](https://user-images.githubusercontent.com/5749884/111899201-f5113980-8a22-11eb-9efe-b2f2c077a733.jpg)
I also noticed on some of the examples of the smoothing by @xmd5a2 that small peaks/troughs had been lost - that seems problematic to me. 

I don't really understand cubic splines, but understand the need to smooth slight deviations along the line of elevation. Thinking about it as a raster image, I was thinking it would work well to apply selective smoothing. e.g. select a 3m range around each contour line, then heavily smooth over this area. I think this would take out the small deviations but leave steep deviations intact. Not sure if that makes sense?
